### ISSUE TITLE
Build is marked as successful even when shell plugin fails.

### ISSUE THREAD

**Issue by clubdeuce** [Block8/PHPCI#1210]

**Comment by dancryer** [comment]
Please can you provide some more information? Screenshots of PHPCI's output, the content of your phpci.yml file, etc?

**Comment by clubdeuce** [comment]
![screen shot 2016-04-27 at 11 19 31 am](https://cloud.githubusercontent.com/assets/2600497/14857408/343a2d60-0c6a-11e6-8ecb-f5c91766da7e.png)

[Gist of phpci.yml](https://gist.github.com/clubdeuce/473c5e83b78e3f30b757ab700a8af757)

**Comment by dancryer** [comment]
OK, I think this is because the shell command that is failing is part of the `success` stage...

If I recall correctly, plugins run as part of `complete`, `success`, `failure`, `broken` or `fixed` do not affect the overall build status as they are not considered to be part of the build itself.

**Comment by clubdeuce** [comment]
Alright, that makes sense.

Then how an I run an rsync command, only for the master branch, and only if the other checks (PHPMD, PHPCS, etc) are successful?

**Comment by dancryer** [comment]
As of v1.7 (tagging the release for this now) - you can run branch-specific plugins like so:

```
branch-master:
    run-option: "after"
    success:
        shell:
            - "rsync ..."
````

That might help?

**Comment by clubdeuce** [comment]
I do not think so, because I need the build to fail if the `rsync` command fails. How can I accomplish that?

**Comment by clubdeuce** [comment]
@dancryer Any thoughts on this?

**Comment by clubdeuce** [comment]
@dancryer No thoughts?
